pds_doi_core/outputs/osti_web_client.py

```diff
@@ -150,6 +151,27 @@ def _verify_osti_reserved_status(self, i_doi_label):
 
         return o_reserved_flag, o_out_text
 
+    def set_doi_fields(self, i_doc, i_doi_fields, i_field_index):
```

Comment:
Maybe I would prefer to have an object to handle that. We can discuss how it is used and how it is different from the record we are creating.

Reply:
This function has been removed.  The TransactionBuilder now handle all things related to parsing things, building transaction, transaction logger.